### DISCUSSION TITLE
Mapbox API key not translating to production build on vercel

### DISCUSSION THREAD

**Discussion by y2katie** [vercel/next.js#22310] [Help]
Hey All! I created a site https://healthhelp.vercel.app/ and the map is showing on my localhost. However, on the deployed site, only the pins are coming through. I built with react and have followed best practices in storing env_variables for that framework. I added the env_var to my settings on vercel but still it will not render properly. Obviously, I am missing a step somewhere, so any help would be appreciated. Let me know if you need anything else from me to help troubleshoot and thank you!!

**Comment by arthurgailes** [comment]
Vercel changes the true URL with each git push, which means manually setting Mapbox API IP Addresses every time you push - an obviously untenable solution. Mapbox also doesn't allow wildcards, so *-USER.vercel.app doesn't work. Would love to know if anybody has solved this or a similar problem.

**Comment by psierra-dev** [comment]
I need help too

**Comment by sadityakumar9211** [comment]
Hey, @y2katie I don't think whatever you're facing is the exact issue as described by @arthurgailes. I was facing the  exact same issue, the map was not rendering but the pins were showing up. I even tried deploying it on a IaaS (Linode) after reading this thread, but it didn't resolve anything. 

From Mapbox Site:
>Mapbox GL JS v2 is distributed as an ES6 compatible JavaScript bundle and is compatible with all major modern browsers.

> The JavaScript bundle is incompatible with some Babel transforms because of the way it shares code between the main thread and Web Worker. We do this to reduce the bundle size and improve rendering performance. If you are using v2 with a module bundler such as Webpack or Rollup along with a transpiler such as Babel, there are three ways to make it compatible:

> - Use browserslist to target transpilation to a set of compatible transforms
> - Explicitly disable transpiling of the Mapbox GL JS bundle
> - Load and transpile Web Worker code separately at the cost of increasing bundle size and reducing performance.

I changed the `package.json` by this (first option):

```json
"browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all",
      "not safari < 10",
      "not chrome < 51",
      "not android < 5",
      "not ie < 12"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
```

After rebuilding the project and deploying again on Linode and Vercel, it worked on both.

**Reply by Yab1** [reply]
This fixed my problem tnx

**Reply by Yab1** [reply]
https://way-finder-plus.vercel.app/

**Reply by huzaifasattar** [reply]
thank you this fixed my problem

**Reply by Aditya-A-Chavan** [reply]
Thank you so much, solved my issue too

**Reply by manjunath-logycent** [reply]
Can confirm that it fixed my problem too, I was using mapbox and was only working in localhost, not in deployment, inspected the network and all api calls were failing without a http status code as a response, checked cors issues, acl policies, privacy options, nothing worked but this did.
Thank you!

**Reply by debbieyuen** [reply]
Thank you so much!! This helped me a lot :)

**Comment by therealltre** [comment]
I recently encountered this same problem with mapbox. I just had to import the environment variables (vercel) and the map rendered.
